libnavigation-core/src/main/java/com/mapbox/navigation/core/NavigationComponentProvider.kt

```diff
@@ -12,13 +12,17 @@ import com.mapbox.navigation.core.internal.trip.service.MapboxTripService
 import com.mapbox.navigation.core.internal.trip.service.TripService
 import com.mapbox.navigation.core.internal.trip.session.MapboxTripSession
 import com.mapbox.navigation.core.trip.session.TripSession
+import com.mapbox.navigation.navigator.internal.MapboxNativeNavigator
+import com.mapbox.navigation.navigator.internal.MapboxNativeNavigatorImpl
 
 internal object NavigationComponentProvider {
     fun createDirectionsSession(
         router: Router
     ): DirectionsSession =
         MapboxDirectionsSession(router)
 
+    fun createNativeNavigator(): MapboxNativeNavigator = MapboxNativeNavigatorImpl
```

Comment:
why would this change any behavior? MapboxNativeNavigatorImpl is still an object?

Reply:
This is for mocking `MapboxNativeNavigator` in tests, native code and JUnit don't get along 😬 Some time back I did this https://github.com/mapbox/mapbox-gl-native/pull/8261/files#diff-20a30dbc2607de725d1e4d09e2bd1403 to workaround the issue. Maybe we can bring it to the Navigation SDK